### DISCUSSION TITLE
Random Timeline Achieved. Unable to render it on the Home Page

### DISCUSSION THREAD

**Discussion by Mitchell-Crumbley** [nss-evening-cohort-14/evening-client-side#310] [Capstones!] [answered]
ISSUE: 
I have created a working component that randomly generates a predetermined number of objects from my showcases array and generates a vertical timeline for those objects. I am unable to call that function in an OnClick for a Button I've created on the Home page. 

What I have searched:
https://trackjs.com/blog/debugging-cannot-read-property-length-of-undefined/
https://medium.com/@gokulraj112/typeerror-cannot-read-property-length-of-undefined-javascript-71120097ee6
https://reactjs.org/docs/handling-events.html

Related ticket:
https://github.com/nss-evening-cohort-14/evening-client-side/discussions/293
-I attempted refactoring my code to mirror this methodology and still came up short. I'm sure it's something small at this point. I'm very close. 

Repos:
Random Array: https://github.com/Mitchell-Crumbley/micd-up/blob/timeline/src/helpers/ArraySort.js
https://github.com/Mitchell-Crumbley/micd-up/blob/timeline/src/views/Home.js

**Comment by drteresavasquez** [comment]
@Mitchell-Crumbley I am not sure as to what I am looking at here. Can you give me the specific line of code that is giving you issues? 

Also, this line is like a return so that is that data you will get back from calling this function: https://github.com/Mitchell-Crumbley/micd-up/blob/7ac6b15f24f655c3cd633bd19563a958ff811738/src/helpers/ArraySort.js#L8

**Reply by drteresavasquez** [reply]
@Mitchell-Crumbley any updates?

**Reply by Mitchell-Crumbley** [reply]
Hey Dr. T!

Trinity met with me at the end of class on Tuesday and helped me understand my issue. The bottom line is I was trying to call a view in an OnClick vs using a component. I believe I am on my way to fixing this. I will post my solution and close the ticket before the end of class!

**Comment by Mitchell-Crumbley** [comment, marked as answer]
Did it! As per usual Trinity was spot on. 

I was making it harder than it needed to be. I separated the function to return a shuffled portion of an array and the timeline it was being passed to. I then simply populate that timeline on the home page instead of a form using setEditing.

Repos:
https://github.com/Mitchell-Crumbley/micd-up/blob/setup/src/components/ShowRandomArray.js
https://github.com/Mitchell-Crumbley/micd-up/blob/setup/src/views/Home.js
